### ISSUE TITLE
self signed cert

### ISSUE THREAD

**Issue by mortenbpost** [redhat-developer/vscode-yaml#107]
Hi,

I'm getting this error after installing the extension. http.proxy and http.proxyStrictSSL is setup.

Unable to load schema from 'https://gist.githubusercontent.com/JPinkney/ccaf3909ef811e5657ca2e2e1fa05d76/raw/f85e51bfb67fdb99ab7653c2953b60087cc871ea/openshift_schema_all.json': Unable to connect to https://gist.githubusercontent.com/JPinkney/ccaf3909ef811e5657ca2e2e1fa05d76/raw/f85e51bfb67fdb99ab7653c2953b60087cc871ea/openshift_schema_all.json through a proxy . Error: self signed certificate in certificate chain

Please advise.

**Comment by jorng** [comment]
The only way I was able to work around this was to download the file locally and provide a `file:///` URL:

```json
    "yaml.schemas": {
        "file:///Users/prentiss/.config/schemas/rio.schema.json": "rio*.yml"
    },
```

**Comment by ReeganExE** [comment]
I work-around by downloading the schema file, add to project and add a settings.json like the following:
```
{
    "yaml.schemas": {
       "./schema.json": "config-*.yml"
    }
}

**Comment by JPinkney** [comment]
I tried looking into this issue a bit. The problem arises for me when I'm trying to access a schema that requires you to be logged into a vpn. I found out that json languageserver resolves the schema correctly but when we run the exact same code in this project we get the self signed cert issue. What's weird is that literally the exact same arguments are passed into the xhr request for request-light but the json languageserver returns the schema but the yaml languageserver errors. I have no idea of vscode is doing something under the hood that makes their request succeed but I can't find anything.

**Comment by JPinkney** [comment]
The only difference I can see is that vscode-json-languagefeatures does their calls to get the schemas from inside of the vscode extension rather than the languageserver.

I tested:
```
xhr({
    url: 'my_uri',
    followRedirects: 5,
    headers,
  }).then(
    (response) => {
      return response.responseText;
    },
    (error: any) => {
      return Promise.reject(error.responseText);
    }
  );
```
inside of the vscode extension and everything resolves correctly. I'm guessing the reason why vscode-json-languagefeatures resolves the schema inside of the vscode extension to avoid this issue. Maybe we will have to adopt that for VSCode in order to get this to work.

**Comment by JPinkney** [comment]
This issue should be fixed in 0.14.0! If there are still any issues please let me know.
